app/controllers/carts_controller.rb

```diff
@@ -0,0 +1,82 @@
+class CartsController < ApplicationController
+  def new
+    authorize Cart
+    @cart = Cart.new
+  end
+
+  def index
+    authorize Cart
+  end
+
+  def create
+    authorize Cart
+    # create a new cart
+    @cart = Cart.new
+
+    # add the user's ID to the new cart
+    @cart.user_id = params['user_id']
```

Comment:
Q1

It is entirely possible to create a new cart for other users than yourself, though I'm not sure if that vulnerability can be used for anything.